src/thor/triplegbuilder.cc

```diff
@@ -92,6 +93,22 @@ void AssignAdmins(const AttributesController& controller,
   }
 }
 
+// Helper function to get the iso country code from an edge
+inline std::string country_code_from_edge(const graph_tile_ptr& tile,
+                                          const valhalla::baldr::DirectedEdge* de) {
+  return tile->admininfo(tile->node(de->endnode())->admin_index()).country_iso();
+}
+
+inline const DirectedEdge*
+get_opposite_de(const DirectedEdge* edge, GraphReader& reader, const graph_tile_ptr& tile) {
```

Comment:
Hmm I thought we had a helper function like this elsewhere.

Same question here though regarding null-pointer check or passing reference.

Reply:
yes we have this as a method on graphreader i would suggest removing it and using that